### DISCUSSION TITLE
Music Videos type has no image fetcher; fixable?

### DISCUSSION THREAD

**Discussion by mharmless** [ankenyr/jellyfin-youtube-metadata-plugin#68] [Q&A] [answered]
Brand new setup of Jellyfin 10.8.4 with YoutubeMetadata 1.0.3.8 on Mint 21. Trying to get it to look good and pull the correct thumbnails with a test data set of 82 videos and accompanying info.json files.

Music Videos is listed as supported, and with the plugin installed I do see the youtube metadata provider on offer with that library type. However, there is no image fetcher and so I end up with white text on a blue background for almost every single video. Unsure why some of them show anything else, but a handful out of the 82 in my test pile do show an image despite having no image fetcher selected.

Using Movies, there is an image fetcher and every video with an info.json gets a proper video thumbnail matching the one on youtube. If you set the view to Thumb or Thumb Card, the appearance matches very well. Movies is a poor choice though as everything gets lumped together and a thousand plus lockpicking lawyer videos would swamp the whole mess. Really want it subdivided by creator.

Using Shows, there are image fetchers for Shows and Episodes, but not Seasons. Which makes sense, youtube doesn't really have seasons. Most episodes get their correct thumbnail, but a few do not and I can't really tell why. Not a fan of the shows option because every creator has exactly one season so it feels like an unnecessary extra screen to get to the content. And a lot of space is spent on the descriptive text, only see four videos at a time which makes for a lot of scrolling and no alternative presentation is on offer to tighten that up. Real TV shows are mostly watched in sequence so this is fine, but for the youtube content I want to archive they are largely self-contained things.

Music Videos divides things up nicely by creator, supports the 'thumb' aspect that matches youtube, and you can view over a dozen thumbnails at once. Just need to get that image fetcher attached to it and good to go. Is there a way to make this happen that I am overlooking?

Also, for my test data set I've redownloaded the videos with embedded metadata _and_ also downloaded info.json files with matching filenames. Is one or the other of these redundant/unneeded?

Thank you for creating this plugin.

**Comment by ankenyr** [comment, marked as answer]
Hi @mharmless I think https://github.com/ankenyr/jellyfin-youtube-metadata-plugin/pull/63 fixes your issue but it isn't where I would want it myself. Once that is submitted or I find the time to fix it myself this should be fixed.

**Reply by mharmless** [reply]
If I'm understanding that correctly, editing line 110 un YTDLImageProvider.cs and recompiling the DLL should do the trick. Thank you, I will grab the source and see if I can make that into reality!
